contracts/amm/AmmOpenSwapService.sol

```diff
@@ -0,0 +1,819 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/IERC20MetadataUpgradeable.sol";
+
+import "../libraries/Constants.sol";
+import "../libraries/math/IporMath.sol";
+import "../libraries/errors/IporErrors.sol";
+import "../libraries/errors/MiltonErrors.sol";
+import "../interfaces/IIporOracle.sol";
+import "../interfaces/IMiltonStorage.sol";
+import "../interfaces/IIporRiskManagementOracle.sol";
+import "../interfaces/IAmmOpenSwapService.sol";
+import "./libraries/types/AmmMiltonTypes.sol";
+import "../libraries/errors/MiltonErrors.sol";
+import "./libraries/IporSwapLogic.sol";
+
+contract AmmOpenSwapService is IAmmOpenSwapService {
+    using Address for address;
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    /// @dev represented in 18 decimals
+    uint256 internal immutable _iporPublicationFee;
```

Comment:
maybe we should have configurations per asset, what do you think?

Reply:
good idea, I will change